src/agent/debuglet.ts

```diff
@@ -112,6 +113,45 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * isReadyManager class is designed to support debug agent in Google Cloud
+ * Function (GCF). As GCF is a serverless environment and we wanted to make
+ * sure debug agent could always capture the snapshots. We designed this class
+ * so that GCF could wait the ListActiveBreakpoitns event completed before
+ * throttling CPU to 0. isReadyManager has two member functions.
+ * isReady is to create a initial promise, or create a promise to replace the
+ * old one if it steals. makeReady will resolve the promise at appropriate
+ * code location (in our case is after listActiveBreakpoints executed).
+ */
+export class IsReadyManager {
+  private promise: Promise<void>;
+  private promiseResolve: (() => void)|null;
+  private promiseResolvedTimestamp: number;
+  constructor() {
+    this.promiseResolvedTimestamp = -Infinity;
```

Comment:
Nit: you can initialize just in line 129 above and delete this constructor.

Reply:
Done.